internal/config-reloader/main.go

```diff
@@ -54,6 +55,12 @@ func main() {
 	}
 
 	configWatcher.startWatch(ctx, updatesChan)
+	if *mode == "one-shot" {
+		logger.Infof("config-reloader one-shot mode completed, exit now")
```

Comment:
If the process[download,parse,write] failed in one-shot mode, should we exit it with code nonzero?

Reply:
Yep, it should exit with non zero code.